gptme/message.py

```diff
@@ -126,39 +124,20 @@ def _content_files_list(
                 )
                 continue
 
-            if provider == "anthropic":
-                content.append(
-                    {
-                        "type": "image",
-                        "source": {
-                            "type": "base64",
-                            "media_type": media_type,
-                            "data": data,
-                        },
-                    }
-                )
-            elif provider == "openai":
-                # OpenAI format
-                content.append(
-                    {
-                        "type": "image_url",
-                        "image_url": {"url": f"data:{media_type};base64,{data}"},
-                    }
-                )
-            else:
-                # Storage/wire format (keep files in `files` list)
-                # Do nothing to integrate files into the message content
-                pass
+            file_content = get_model().manager.prepare_file(media_type, data)
+            if file_content:
+                content.append(file_content)
```

Comment:
This is not valid, you cannot rely in get_model() returning the correct model for a given call to `to_dict` or `_content_files_list`.

Reply:
Left some comments about this on the rest of the PR, we can make it work!